### PR TITLE
mgr/cephadm: simplify and improve placement

### PR DESCRIPTION
**Pull request by liewegas**
- Make placement simpler so that its behavior can match rook
- Make placement optional

### CODE REVIEW

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
green http://pulpito.ceph.com/sage-2020-03-09_05:21:40-rados:cephadm-wip-sage-testing-2020-03-08-2121-distro-basic-smithi/

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2359,7 +2359,23 @@ def add_mgr(self, spec):
         return self._add_daemon('mgr', spec, self._create_mgr)
 
     def _apply(self, spec):
-        self.log.info('Saving service %s spec' % spec.service_name())
+        if spec.placement.is_empty():
+            # fill in default placement
+            defaults = {
+                'mon': orchestrator.PlacementSpec(count=5),
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: As long as this works, if we have two or 4 hosts registered!

Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: Yep, it works.

Right now, though, it silently degrades.  I wonder if in the future we want to have some health warning in these cases?

**Pull request merged**